From 4989e8e0fd1b8d7f7b747e8557f03ca1419a52e0 Mon Sep 17 00:00:00 2001
From: StanleyYP Wang <StanleyYP.Wang@mediatek.com>
Date: Thu, 4 Sep 2025 16:51:49 +0800
Subject: [PATCH 119/128] mtk: mac80211: disable beacon in start ap for mld ap

When beacon protection is enabled, set beacon will be called in
__ieee80211_set_default_beacon_key, which results in beaconing
being enabled before all the links are ready.

Signed-off-by: StanleyYP Wang <StanleyYP.Wang@mediatek.com>
---
 net/mac80211/cfg.c | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/net/mac80211/cfg.c b/net/mac80211/cfg.c
index 8428ef03..caf19f3b 100644
--- a/net/mac80211/cfg.c
+++ b/net/mac80211/cfg.c
@@ -1736,11 +1736,6 @@ static int ieee80211_start_ap(struct wiphy *wiphy, struct net_device *dev,
 
 	ieee80211_vif_cfg_change_notify(sdata, BSS_CHANGED_SSID);
 	ieee80211_link_info_change_notify(sdata, link, changed);
-	/* for MLD AP, enable_beacon is false during the first beacon set,
-	 * enable it after that. This allows userspace to control the
-	 * beacon enable timing.
-	 */
-	link_conf->enable_beacon = true;
 
 	if (ieee80211_num_beaconing_links(sdata) <= 1)
 		netif_carrier_on(dev);
@@ -1815,6 +1810,7 @@ static int ieee80211_change_beacon(struct wiphy *wiphy, struct net_device *dev,
 		changed |= BSS_CHANGED_HE_BSS_COLOR;
 	}
 
+	link_conf->enable_beacon = true;
 	ieee80211_link_info_change_notify(sdata, link, changed);
 	return 0;
 }
-- 
2.45.2

